components/match2/commons/match_group_util.lua

```diff
@@ -81,13 +81,33 @@ MatchGroupUtil.types.BracketData = TypeUtil.union(
 	MatchGroupUtil.types.BracketBracketData
 )
 
+---@class standardPlayer
+---@field displayName string?
+---@field flag string?
+---@field pageName string?
+---@field team string?
 MatchGroupUtil.types.Player = TypeUtil.struct({
 	displayName = 'string?',
 	flag = 'string?',
 	pageName = 'string?',
 	team = 'string?',
 })
 
+---@class standardOpponent
+---@field advanceBg string?
+---@field advances boolean?
+---@field icon string?
+---@field name string?
+---@field placement number?
+---@field placement2 number?
+---@field players standardPlayer[]
+---@field score number?
+---@field score2 number?
+---@field status string?
+---@field status2 string?
+---@field template string?
+---@field type string
+---@field team string?
```

Comment:
Feels like this is a `MatchOpponent` rather than a `StandardOpponent`? A MatchOpponent being a child (ie extension) of a StandardOpponent?

Reply:
well we convert into or from this struct in quite a few places outside of match2
- converting from/to lpdb_placement
- usually we convert to this whenever want to use opponent display (e.g. new portal players)

When i opened this PR I thought about spliting it up, but in some places they get intermingeled.
It basically would mean we would have to allow both opponent types (match and standard) for almost all (except BracketOpponentEntry class), else we would get anno warnings for type mismatch all over the place

Reply:
Ok